appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.java

```diff
@@ -451,11 +482,48 @@ private void handleSlideChanged(Fragment oldFragment, Fragment newFragment) {
 
     /**
      * Called after a new slide has been selected
+     * {@link #canChangeLastSlide(int)} is called to check if the last slide can be changed.
      *
      * @param position Position of the new selected slide
      */
     protected void onPageSelected(int position) {
-        // Empty method
+        LogHelper.d(TAG, "Selected slide number: " + (position + 1));
+        canChangeLastSlide(position);
+    }
+
+
+    /**
+     * Called by {@link #onPageSelected(int)} to check if the user has requested permissions on the previous slide.
+     * This method is required as a safety check because otherwise,
+     * the user might skip a slide containing permissions.
+     * <p>
+     * This method will send the user back to the first slide that contains permissions.
+     *
+     * @param position - The position that the user has changed to.
+     *                 <p>
+     *                 TODO(1): This method will require tweaking when switching to Viewpager2 as
+     *                 there is no RTL support in the current viewpager and that messes up the position of slides.
+     *                 <p>
+     *                 If the position issue is resolved in Viewpager2, simply remove the code inside if(isRtl()).
+     */
+    private void canChangeLastSlide(int position) {
```

Comment:
Naming feels wrong to me. Seeing `canChangeLastSlide` is assume the method would return a `boolean`. Also this method is not _checking_ if I can Change to the last slide, but doing other stuff.

Reply:
Done